src/Symfony/Component/Serializer/Tests/Normalizer/AbstractNormalizerTest.php

```diff
@@ -247,6 +248,19 @@ public static function getNormalizer()
         yield [new ObjectNormalizer(null, null, null, $extractor)];
     }
 
+    public function testVariadicConstructorDenormalization()
+    {
+        $data = '{"foo":"woo","baz":[{"foo":null,"bar":null,"baz":null,"qux":null},{"foo":null,"bar":null,"baz":null,"qux":null}]}';
+        $expected = new DummyWithWithVariadicParameterConstructor('woo', 1, new Dummy(), new Dummy());
+
+        $normalizer = new ObjectNormalizer();
+        $serializer = new Serializer([$normalizer], [new JsonEncoder()]);
+        $normalizer->setSerializer($serializer);
+
+        $actual = $normalizer->denormalize(json_decode($data, true), DummyWithWithVariadicParameterConstructor::class);
+        $this->assertEquals($expected, $actual);
```

Comment:
Your test can be simplified a bit by:
- not using JSON string but a normalized array
- not instantiating a `JsonEncoder` for nothing

Eg:
```php
$data = [
    'foo' => 'woo',
    'baz' => [
        ['foo' => null, 'bar' => null, 'baz' => null, 'qux' => null],
        ['foo' => null, 'bar' => null, 'baz' => null, 'qux' => null],
    ],
];

$normalizer = new ObjectNormalizer();
$normalizer->setSerializer(new Serializer([$normalizer]));

$expected = new DummyWithWithVariadicParameterConstructor('woo', 1, new Dummy(), new Dummy());
$actual = $normalizer->denormalize($data, DummyWithWithVariadicParameterConstructor::class);

$this->assertEquals($expected, $actual);
```